crates/gpio/src/backend.rs

```diff
@@ -110,6 +136,11 @@ impl TryFrom<&str> for DeviceConfig {
                     GpioDeviceConfig::PhysicalDevice(devid)
                 }
             })?;
+            #[cfg(not(feature = "dummy"))]
```

Comment:
Instead of this, maybe we can just do this:

                _ => {
                   #[cfg(not(feature = "dummy"))]
                   panic!("mock implementatoins aren't supported");

                    let devid = info.parse::<u32>().map_err(Error::ParseFailure)?;
                    GpioDeviceConfig::PhysicalDevice(devid)
                }

Reply:
Now we do:

```
            match cfg {
                GpioDeviceType::PhysicalDevice(device_num) => {
                    let controller = PhysDevice::open(device_num).map_err(Error::CouldNotOpenDevice)?;
                    start_device_backend(controller, socket.clone())?;
                }
                #[cfg(feature = "mock_gpio")]
                GpioDeviceType::SimulatedDevice(num_gpios) => {
                    let controller = MockGpioDevice::open(num_gpios).map_err(Error::CouldNotOpenDevice)?;
                    start_device_backend(controller, socket.clone())?;
                }
                #[cfg(not(feature = "mock_gpio"))]
                _ => {
                    return Err(Error::DeviceUnimplemented);
                }
            };
```